docs/dev/topology-over-raft.md

```diff
@@ -218,3 +218,60 @@ There are also a few static columns for cluster-global properties:
 - `unpublished_cdc_generations` - the IDs of the committed yet unpublished CDC generations
 - `global_topology_request` - if set, contains one of the supported global topology requests
 - `new_cdc_generation_data_uuid` - used in `commit_cdc_generation` state, the UUID of the generation to be committed
+
+# Join procedure
+
+In topology on raft mode, new nodes need to go through a new handshake procedure
+before they can join the cluster, including joining group 0. The handshake
+happens during raft discovery and replaced the old `GROUP0_MODIFY_CONFIG`.
+
+Two RPC verbs are introduced:
+
+- `JOIN_NODE_REQUEST`
+  Sent by the node that wishes to join the cluster. It contains some basic
+  information about the node that will have to be verified by the coordinator.
+
+- `JOIN_NODE_REPLY`
+  Sent by the topology coordinator back to the joining node. It contains
+  coordinator's decision about the node - whether the request was accepted
+  or rejected.
+
+The procedure is designed to be idempotent. In case the joining node crashes
```

Comment:
So I wonder if we should try to support the scenario where bootstrapping node restarts in the middle. I tested what happens in this case with the gossiper and it looks like the scenario is not supported. If the node restart (at least while streaming) the next bootstrap attempt will fails with an error that says that the node is already in the cluster and removenode should be used before trying the bootstrap once more. @asias?

Reply:
@gleb-cloudius you will probably get different result if you wait for 2 minutes before retrying the bootstrap

Reply:
I know. Because the node will be flushed from the gossiper. But is this really what we want to emulate? If an admin had to know to wait 2 minutes after failure he may as well know to remove the data dir and start new bootstrap immediately.

Reply:
Agreed